randovania/games/samus_returns/generator/bootstrap.py

```diff
@@ -14,21 +15,31 @@
     from randovania.game_description.game_patches import GamePatches
     from randovania.game_description.resources.resource_database import ResourceDatabase
     from randovania.game_description.resources.resource_info import ResourceGain
+    from randovania.games.samus_returns.layout.msr_configuration import MSRArtifactConfig
     from randovania.generator.pickup_pool import PoolResults
     from randovania.layout.base.base_configuration import BaseConfiguration
 
 
 def all_dna_locations(game: GameDescription, config: MSRArtifactConfig) -> list[PickupNode]:
-    locations: list[PickupNode] = []
+    locations = []
 
     for node in game.region_list.all_nodes:
         if isinstance(node, PickupNode):
-            # Metroid pickups
             pickup_type = node.extra.get("pickup_type")
-            if config.prefer_metroids and pickup_type is not None and pickup_type == "metroid":
+            pickup_index = node.pickup_index.index
+            # Metroid pickups
+            if pickup_type == "metroid":
+                if config.prefer_metroids and config.prefer_stronger_metroids:
+                    locations.append(node)
+                elif config.prefer_metroids and pickup_index not in _stronger_metroid_indices:
+                    locations.append(node)
+                elif config.prefer_stronger_metroids and pickup_index in _stronger_metroid_indices:
+                    locations.append(node)
+            # Boss pickups
+            elif config.prefer_bosses and pickup_index in _boss_indices:
                 locations.append(node)
             # DNA anywhere
-            elif not config.prefer_metroids:
+            elif not config.prefer_metroids and not config.prefer_stronger_metroids and not config.prefer_bosses:
```

Comment:
I think this was "wrong" at the first place because isn't this just `else:`? There shouldn't be another case beause it's already ensured that it is a `PickupNode` and all other configuration options are handled by the cases above.

Reply:
Without this, it would incorrectly place dna when I had an option selected (I had bosses only enabled and it place it anywhere)

Reply:
Oops, yes you are right. If you would use `else` and you iterate through all nodes and only have boss pickups and then of course often no condition is true and the else would catch it and add the node.
So, nvm!